mongoOperator/models/BaseModel.py

```diff
@@ -55,7 +55,7 @@ def __eq__(self, other: any) -> bool:
         :param other: The other object.
         :return: True if they are equal, False otherwise.
         """
-        return type(other) == type(self) and other.to_dict() == self.to_dict()
+        return isinstance(other, type(self)) and other.to_dict() == self.to_dict()
```

Comment:
Shouldn't the types be exactly the same?

Reply:
The previous check resulted in:
mongoOperator/models/BaseModel.py:58:15: C0123: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)